addons/hr_recruitment/models/hr_applicant.py

```diff
@@ -504,17 +565,29 @@ def message_new(self, msg, custom_values=None):
         # do not want to explicitly set user_id to False; however we do not
         # want the gateway user to be responsible if no other responsible is
         # found.
+        job_offer_parsers = {
+                'LinkedIn': self._parse_linkedin_application,
+                'ICTJobs': self._parse_ict_job_application,
+            }
         self = self.with_context(default_user_id=False)
         stage = False
         if custom_values and 'job_id' in custom_values:
             stage = self.env['hr.job'].browse(custom_values['job_id'])._get_first_stage()
+
         val = msg.get('from').split('<')[0]
-        defaults = {
-            'name': msg.get('subject') or _("No Subject"),
-            'partner_name': val,
-            'email_from': msg.get('from'),
-            'partner_id': msg.get('author_id', False),
-        }
+
+        msg_from = msg.get('from')
+        job_website = next((value for key, value in JOB_POSTING_SERVICES_MAILS.items() if key in msg_from), None)
+
+        if job_website:
+            defaults = job_offer_parsers[job_website](msg)
```

Comment:
In the defaults below they use msg.get('subject') or _("No Subject"). Did you not use that fallback because we know for sure that the message from these services will always have a subject?